src/Core/Content/Media/Core/Application/RemoteThumbnailLoader.php

```diff
@@ -168,25 +172,37 @@ private function getBaseUrl(): string
         return \rtrim($this->filesystem->publicUrl(''), '/');
     }
 
-    private function getUrl(string $mediaUrl, string $mediaPath, string $width, string $height, ?\DateTimeInterface $mediaUpdatedAt): string
+    private function getUrl(Entity|PartialEntity|MediaEntity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
```

Comment:
```suggestion
    private function getUrl(Entity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
```

as you allow `Entity`, you don't need to add the child types